compiler-rt/lib/sanitizer_common/sanitizer_procmaps_mac.cpp

```diff
@@ -445,6 +443,56 @@ bool MemoryMappingLayout::Next(MemoryMappedSegment *segment) {
   return false;
 }
 
+// NOTE: Verify expects to be called immediately after Reset(), since otherwise
+// it may miss some mappings. Verify will Reset() the layout after verification.
+bool MemoryMappingLayout::Verify() {
```

Comment:
Can you make it a static function, with parameters of what is needed
and avoid Verify in the header?

Unless there is a plan to extend it to other platforms?